File: GuardianMonsters/media/src/de/limbusdev/guardianmonsters/media/NullMediaManager.java
package de.limbusdev.guardianmonsters.media;

import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.Animation;
import com.badlogic.gdx.graphics.g2d.TextureAtlas;
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion;
import com.badlogic.gdx.graphics.g2d.TextureRegion;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.utils.ArrayMap;

import de.limbusdev.guardianmonsters.enums.SkyDirection;
import de.limbusdev.guardianmonsters.scene2d.AnimatedImage;


/**
 * NullMedia, gets returned from Service Locator, when no Media Service has been injected yet.
 * Created by georg on 14.11.16.
 */

public class NullMediaManager implements IMediaManager
{

    public void dispose() {}

    public TextureAtlas getTextureAtlasType(TextureAtlasType type) {
        return null;
    }

    public TextureAtlas getPersonTextureAtlas(boolean male, int index) {
        return null;
    }

    @Override
    public ArrayMap<SkyDirection, Animation<AtlasRegion>> getPersonAnimationSet(boolean gender, int index) {
        return null;
    }

    @Override
    public ArrayMap<SkyDirection, Animation<AtlasRegion>> getPersonAnimationSet(String name) {
        return null;
    }

    @Override
    public ArrayMap<SkyDirection, Animation<AtlasRegion>> getPersonAnimationSet(TextureAtlas atlas) {
        return null;
    }

    @Override
    public Texture getTexture(String path) {
        return null;
    }

    public TextureAtlas.AtlasRegion getMonsterSprite(int index) {
        return null;
    }

    @Override
    public TextureAtlas.AtlasRegion getMonsterMiniSprite(int index) {
        return null;
    }

    @Override
    public Image getMonsterFace(int id) {
        return null;
    }

    public TextureRegion getBackgroundTexture(int index) {
        return null;
    }

    @Override
    public TextureAtlas getTextureAtlas(String path) {
        return null;
    }

    public Animation getAttackAnimation(String attack) {
        return null;
    }

    public Animation getTileAnimation(int index) {
        return null;
    }

    public Animation getObjectAnimation(String id) {return null;}

    @Override
    public Image getMetamorphosisBackground() {
        return null;
    }

    @Override
    public AnimatedImage getMetamorphosisAnimation() {
        return null;
    }

}

File: GuardianMonsters/media/src/de/limbusdev/guardianmonsters/media/NullMediaManager.kt
package de.limbusdev.guardianmonsters.media

import com.badlogic.gdx.graphics.Pixmap
import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Animation
import com.badlogic.gdx.graphics.g2d.TextureAtlas
import com.badlogic.gdx.graphics.g2d.TextureAtlas.AtlasRegion
import com.badlogic.gdx.graphics.g2d.TextureRegion
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.utils.Drawable
import com.badlogic.gdx.scenes.scene2d.utils.TextureRegionDrawable
import com.badlogic.gdx.utils.ArrayMap

import de.limbusdev.guardianmonsters.enums.SkyDirection
import de.limbusdev.guardianmonsters.scene2d.AnimatedImage


/**
 * NullMedia, gets returned from Service Locator, when no Media Service has been injected yet.
 *
 * @author Georg Eckert
 */

class NullMediaManager : IMediaManager
{
    // .......................................................................... Textures & Sprites
    override fun getTexture(path: String) = Texture(100, 100, Pixmap.Format.Alpha)

    override fun getMonsterSprite(index: Int, form: Int) : TextureRegion = TextureRegion()

    override fun getMonsterMiniSprite(index: Int, form: Int) : TextureRegion = TextureRegion()

    override fun getItemDrawable(nameID: String) : Drawable = TextureRegionDrawable()

    override fun getMonsterFace(id: Int, form: Int): Image = Image()

    override fun getBackgroundTexture(index: Int): TextureRegion = TextureRegion()

    override fun getMetamorphosisBackground(): Image = Image()


    // ....................................................................................... Atlas
    override fun getTextureAtlas(path: String): TextureAtlas = TextureAtlas()

    override fun getTextureAtlasType(type: TextureAtlasType): TextureAtlas = TextureAtlas()

    override fun getPersonTextureAtlas(male: Boolean, index: Int): TextureAtlas = TextureAtlas()


    // .................................................................................. Animations
    override fun getMetamorphosisAnimation(): AnimatedImage = AnimatedImage()

    override fun getSummoningAnimation(): Animation<TextureRegion> = Animation(1f)

    override fun getBanningAnimation(): Animation<TextureRegion> = Animation(1f)

    override fun getAttackAnimation(attack: String): Animation<TextureRegion> = Animation(1f)

    override fun getStatusEffectAnimation(statusEffect: Enum<*>): Animation<TextureRegion> = Animation(1f)


    // .............................................................................. Map Animations
    override fun getPersonAnimationSet(gender: Boolean, index: Int): ArrayMap<SkyDirection, Animation<TextureRegion>>
    {
        return ArrayMap()
    }

    override fun getPersonAnimationSet(name: String): ArrayMap<SkyDirection, Animation<TextureRegion>>
    {
        return ArrayMap()
    }

    override fun getPersonAnimationSet(atlas: TextureAtlas): ArrayMap<SkyDirection, Animation<TextureRegion>>
    {
        return ArrayMap()
    }

    override fun getTileAnimation(index: Int): Animation<TextureRegion> = Animation(1f)

    override fun getObjectAnimation(id: String): Animation<TextureRegion> = Animation(1f)


    // .................................................................................. Management
    override fun dispose() {}
}
